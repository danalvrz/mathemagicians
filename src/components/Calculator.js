import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calculatorCointainer">
        <div className="resultCointainer">
          <p className="result">0</p>
        </div>
        <div className="buttons">
          <div className="bttn">AC</div>
          <div className="bttn">+/-</div>
          <div className="bttn">%</div>
          <div className="bttn operator">รท</div>
          <div className="bttn">7</div>
          <div className="bttn">8</div>
          <div className="bttn">9</div>
          <div className="bttn operator">x</div>
          <div className="bttn">4</div>
          <div className="bttn">5</div>
          <div className="bttn">6</div>
          <div className="bttn operator">-</div>
          <div className="bttn">1</div>
          <div className="bttn">2</div>
          <div className="bttn">3</div>
          <div className="bttn operator">+</div>
          <div className="bttn zero">0</div>
          <div className="bttn">.</div>
          <div className="bttn operator">=</div>
        </div>
      </div>
    );
  }
}

export default Calculator;
