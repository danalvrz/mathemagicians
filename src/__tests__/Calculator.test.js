import React from 'react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line no-unused-vars
import { render, fireEvent } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import Calculator from '../components/Calculator';

test('renders correctly', () => {
  const tree = renderer.create(
    <Router>
      <Calculator />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
