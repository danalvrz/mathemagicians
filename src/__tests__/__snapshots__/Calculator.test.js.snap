// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorTesting renders correctly 1`] = `
<div>
  <header
    className="menuContainer"
  >
    <h1>
      Mathemagicians
    </h1>
    <nav
      className="menuList"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
      |
      <a
        href="#/calculator"
        onClick={[Function]}
      >
        Cowculator
      </a>
      |
      <a
        href="#/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </nav>
  </header>
  <div
    className="cowculator"
  >
    <span
      className="cowculatorLegend"
      style={
        Object {
          "color": "aliceblue",
        }
      }
    >
      Let's get crunchin'!
      <br />
      moooooooooo!
    </span>
    <div
      className="calculatorCointainer"
    >
      <div
        className="resultCointainer"
        style={
          Object {
            "color": "aliceblue",
          }
        }
      >
        <p
          className="result"
          id="resultDisplay"
        >
           
           
        </p>
      </div>
      <div
        className="buttons"
      >
        <button
          className="bttn operator"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="bttn "
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="bttn "
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="bttn operator"
          name="÷"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
        <button
          className="bttn "
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="bttn "
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="bttn "
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="bttn operator"
          name="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="bttn "
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="bttn "
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="bttn "
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="bttn operator"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="bttn "
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="bttn "
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="bttn "
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="bttn operator"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="bttn zero"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="bttn "
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="bttn operator"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
